### PR TITLE
feat: monitoring gc performance with signposts

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

**Review by projedi — APPROVED** [review]

File: docs/topics/native/native-memory-manager.md [hunk 1]
```diff
@@ -31,15 +31,34 @@ and waits for its completion.
...
 
 ```none
 -Xruntime-logs=gc=info
 ```
 
 Currently, the logs are only printed to `stderr`.
 
+On Apple platforms, you can take advantage of the Xcode Instruments toolkit to debug iOS app performance.
+The garbage collector reports pauses with signposts available in Instruments.
+Signposts allow for custom logging within your app, so you can check if a GC pause corresponds to the application freeze.
```
Comment (koshachy) [inline docs/topics/native/native-memory-manager.md:45]: Suggestion (up to you):

Signposts enable custom logging within your app, allowing you to check if a GC pause causes an application freeze.

File: docs/topics/native/native-memory-manager.md [hunk 2]
```diff
@@ -31,15 +31,34 @@ and waits for its completion.
...
+
+To track GC-related pauses in your app:
+
+1. Open Xcode, go to **Product** | **Profile** or press <shortcut>Cmd + I</shortcut>. This will compile your app and
+   launch Instruments.
```
Comment (koshachy) [inline docs/topics/native/native-memory-manager.md:50]:
```suggestion
1. Open Xcode, go to **Product** | **Profile** or press <shortcut>Cmd + I</shortcut>. This action compiles your app and
   launches Instruments.
```

File: docs/topics/native/native-memory-manager.md [hunk 3]
```diff
@@ -31,15 +31,34 @@ and waits for its completion.
...
+2. In the template selection, select **os_signpost**.
+3. Configure it by specifying `org.kotlinlang.native.runtime` as **subsystem** and `safepoint` as **category**.
+4. Click the red record button to run your app and start recording signpost events.
```
Comment (koshachy) [inline docs/topics/native/native-memory-manager.md:53]:
```suggestion
4. Click the red record button to run your app and start recording signpost events:
```

**Review by koshachy — APPROVED** [review]
Minor quibbles and suggestions.
Take a look. Accepting them is up to you.

**Pull request merged**